barretenberg/cpp/src/barretenberg/commitment_schemes/commit.bench.cpp

```diff
@@ -7,25 +7,146 @@ namespace bb {
 
 template <typename Curve> std::shared_ptr<CommitmentKey<Curve>> create_commitment_key(const size_t num_points)
 {
+    bb::srs::init_crs_factory("../srs_db/ignition");
     std::string srs_path;
     return std::make_shared<CommitmentKey<Curve>>(num_points);
 }
 
-constexpr size_t MAX_LOG_NUM_POINTS = 24;
-constexpr size_t MAX_NUM_POINTS = 1 << MAX_LOG_NUM_POINTS;
+// Generate a polynomial with a specified number of nonzero random coefficients
+template <typename FF> Polynomial<FF> sparse_random_poly(const size_t size, const size_t num_nonzero)
+{
+    auto& engine = numeric::get_debug_randomness();
+    auto polynomial = Polynomial<FF>(size);
+
+    for (size_t i = 0; i < num_nonzero; i++) {
+        size_t idx = engine.get_random_uint32() % size;
+        polynomial[idx] = FF::random_element();
+    }
 
-auto key = create_commitment_key<curve::BN254>(MAX_NUM_POINTS);
+    return polynomial;
+}
+
+constexpr size_t MIN_LOG_NUM_POINTS = 16;
+constexpr size_t MAX_LOG_NUM_POINTS = 20;
+constexpr size_t MAX_NUM_POINTS = 1 << MAX_LOG_NUM_POINTS;
```

Comment:
FWIW we prob do care about 2^24 since that's the ClientIVC recursive verifier size right now.

Reply:
yeah easy enough to change for an interested party but I was running these a lot and anything larger was taking too long